### PR TITLE
Be gentler in setting BinaryBuilder makevars

### PR DESCRIPTION
**Pull request by staticfloat**
Many users have expressed the frustration that setting `USE_BINARYBUILDER=0` in `Make.user` doesn't take; this is because the initializations here should have been `?=` (e.g. "set unless already set") instead of `:=` (e.g. "set immediately").  This remedies that.

### CODE REVIEW

File: Make.inc [hunk 1]
```diff
@@ -982,17 +982,17 @@ endif
...
+USE_BINARYBUILDER ?= 0
 endif
 
 # This is the set of projects that BinaryBuilder dependencies are hooked up for.
 BB_PROJECTS := OPENBLAS LLVM SUITESPARSE OPENLIBM GMP MBEDTLS LIBSSH2 MPFR CURL LIBGIT2 PCRE LIBUV UNWIND DSFMT OBJCONV
-$(foreach proj,$(BB_PROJECTS),$(if $(USE_BINARYBUILDER_$(proj)),,$(eval USE_BINARYBUILDER_$(proj) := $(USE_BINARYBUILDER))))
+$(foreach proj,$(BB_PROJECTS),$(if $(USE_BINARYBUILDER_$(proj)),,$(eval USE_BINARYBUILDER_$(proj) ?= $(USE_BINARYBUILDER))))
 
 # Use the Assertions build
 BINARYBUILDER_LLVM_ASSERTS := 0
```
Comment (vchuravy) [inline Make.inc]:
```suggestion
BINARYBUILDER_LLVM_ASSERTS ?= 0
```

**Review by vchuravy — APPROVED** [review]

File: Make.inc [hunk 2]
```diff
@@ -982,20 +982,20 @@ endif
...
+USE_BINARYBUILDER ?= 0
 endif
 
 # This is the set of projects that BinaryBuilder dependencies are hooked up for.
 BB_PROJECTS := OPENBLAS LLVM SUITESPARSE OPENLIBM GMP MBEDTLS LIBSSH2 MPFR CURL LIBGIT2 PCRE LIBUV UNWIND DSFMT OBJCONV
-$(foreach proj,$(BB_PROJECTS),$(if $(USE_BINARYBUILDER_$(proj)),,$(eval USE_BINARYBUILDER_$(proj) := $(USE_BINARYBUILDER))))
+$(foreach proj,$(BB_PROJECTS),$(if $(USE_BINARYBUILDER_$(proj)),,$(eval USE_BINARYBUILDER_$(proj) ?= $(USE_BINARYBUILDER))))
 
 # Use the Assertions build
-BINARYBUILDER_LLVM_ASSERTS := 0
```
Comment (Keno) [inline Make.inc]: Unrelated, but shouldn't this just look at the `LLVM_ASSERTIONS` variable?

Comment (staticfloat) [reply Make.inc]: Probably?  @vchuravy is that just a silly oversight on my part?

Comment (vchuravy) [reply Make.inc]: I originally had it split out to make switching more explicit. No strong feelings on my part.

**Comment by StefanKarpinski** [comment]
I've thought this would be good to change for a long time. Would also be good to change any other variables that can be set in Make.user to configure the build process so that `override` isn't needed.

**Comment by staticfloat** [comment]
Will address the LLVM_ASSETS stuff in a future PR.

**Pull request merged**